native/core/src/models/status/room_state.rs

```diff
@@ -0,0 +1,1513 @@
+use matrix_sdk_base::ruma::events::{
+    policy::rule::{
+        room::{PolicyRuleRoomEventContent, PossiblyRedactedPolicyRuleRoomEventContent},
+        server::{PolicyRuleServerEventContent, PossiblyRedactedPolicyRuleServerEventContent},
+        user::{PolicyRuleUserEventContent, PossiblyRedactedPolicyRuleUserEventContent},
+    },
+    room::{
+        aliases::{PossiblyRedactedRoomAliasesEventContent, RoomAliasesEventContent},
+        avatar::RoomAvatarEventContent,
+        canonical_alias::RoomCanonicalAliasEventContent,
+        create::RoomCreateEventContent,
+        encryption::{PossiblyRedactedRoomEncryptionEventContent, RoomEncryptionEventContent},
+        guest_access::{PossiblyRedactedRoomGuestAccessEventContent, RoomGuestAccessEventContent},
+        history_visibility::RoomHistoryVisibilityEventContent,
+        join_rules::RoomJoinRulesEventContent,
+        name::{PossiblyRedactedRoomNameEventContent, RoomNameEventContent},
+        pinned_events::{
+            PossiblyRedactedRoomPinnedEventsEventContent, RoomPinnedEventsEventContent,
+        },
+        power_levels::RoomPowerLevelsEventContent,
+        server_acl::RoomServerAclEventContent,
+        third_party_invite::{
+            PossiblyRedactedRoomThirdPartyInviteEventContent, RoomThirdPartyInviteEventContent,
+        },
+        tombstone::{PossiblyRedactedRoomTombstoneEventContent, RoomTombstoneEventContent},
+        topic::{PossiblyRedactedRoomTopicEventContent, RoomTopicEventContent},
+    },
+    space::{
+        child::{PossiblyRedactedSpaceChildEventContent, SpaceChildEventContent},
+        parent::{PossiblyRedactedSpaceParentEventContent, SpaceParentEventContent},
+    },
+};
+use serde::{Deserialize, Serialize};
+
+use crate::util::do_vecs_match;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum OtherState {
```

Comment:
I don't understand the point of this enum. Next to the bad name (it should probably be `RoomState` not just `OtherState`), it seems to be only used where we already have an enum of different states as an additional wrapper that is the unwrapped.